src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -574,6 +574,34 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
     val runsRightAfter = None
   } with GenBCode
 
+  // phaseName = "analyzeDeps"
+  object analyzeDeps extends {
+    val global: Global.this.type = Global.this
+  } with SubComponent {
+    val phaseName = "analyzeDeps"
+    val runsAfter = List("jvm")
+    val runsRightAfter = None
+
+    def newPhase(prev: Phase): GlobalPhase = {
+      new AnalyzeDepsPhase(prev)
+    }
+    private class AnalyzeDepsPhase(prev: Phase) extends GlobalPhase(prev) {
```

Comment:
I don't remember details since my experiments took place last December. I think my problem was in traversing the entire symbol table, which is much-much bigger than just the top-level part.